sonaranalyzer-dotnet/src/SonarAnalyzer.Common/SonarAnalyzer.Common.csproj

```diff
@@ -31,13 +33,25 @@
     <AdditionalFiles Include="..\SonarLint.xml" Link="Properties\SonarLint.xml" />
   </ItemGroup>
 
+  <PropertyGroup>
+    <!-- The path to the ucfg submodule, relative to this project -->
+    <UcfgProtoPath>./../../../sonar-ucfg/src/main/protobuf/</UcfgProtoPath>
+  </PropertyGroup>
+
   <ItemGroup>
-    <None Include="..\..\..\sonar-ucfg\src\main\protobuf\ucfg.proto" Link="Protobuf\ucfg.proto" />
+    <None Include="$(UcfgProtoPath)ucfg.proto" Link="Protobuf\ucfg.proto" />
   </ItemGroup>
 
-  <Target Name="GenerateProtobufClasses" BeforeTargets="PrepareForBuild" Outputs="Protobuf\AnalyzerReport.cs,Protobuf\Ucfg.cs" Inputs="Protobuf\AnalyzerReport.proto,..\..\..\sonar-ucfg\src\main\protobuf\ucfg.proto">
-    <Message Importance="high" Text="Generating protobuf class..." />
-    <Exec WorkingDirectory="Protobuf" Command="build.bat" />
+  <!-- Generate protobuf classes, supports windows and linux execution -->
+  <Target Name="GenerateProtobufClasses" BeforeTargets="CoreCompile" Outputs="./Protobuf/Ucfg.cs;.\Protobuf\AnalyzerReport.cs" Inputs="$(UcfgProtoPath)ucfg.proto;.\Protobuf\AnalyzerReport.proto">
+    <Message Importance="high" Text="Generating protobuf classes..." />
+    <PropertyGroup>
+      <!-- This path is platform dependent, it should become conditional when we decide we want to build on Linux or OSX -->
+      <ProtocExe>"$(UserProfile)/.nuget/packages/Google.Protobuf.Tools/3.4.0/tools/windows_x64/protoc.exe"</ProtocExe>
```

Comment:
In SonarSecurity this is multi-platform, I removed the rest of the code because we don't plan to build on Linux so far...

Reply:
JulienH was trying to build on Linux last week and this was the main thing that stopped him. If we already have a cross-plat version that works we should use it here too.

Reply:
it won't work for the VSIX project as far as I could imagine... I will copy the cross-plat code here, though.